python/cuxfilter/charts/core/non_aggregate/core_non_aggregate.py

```diff
@@ -109,6 +111,8 @@ def selection_callback(xmin, xmax, ymin, ymax):
             temp_data = dashboard_cls._query(
                 dashboard_cls._generate_query_str()
             )
+            if len(temp_data) == 0:
+                temp_data = cudf.DataFrame({k: cp.nan for k in self.source})
```

Comment:
Would it be possible to handle zero-len inside `_reload_charts` so that the caller does not have to care about this special case?

Reply:
fixed this. Since this is only required for non_aggregate(datashader charts), moved the DF check over to datashader reload_chart for the charts requiring it (datashader - non_aggregate, graph, stacked_lines)